app/uk/gov/hmrc/customs/emailfrontend/controllers/WhatIsYourEmailController.scala

```diff
@@ -90,7 +74,7 @@ class WhatIsYourEmailController @Inject()(actions: Actions, view: change_your_em
         }
       },
       formData => {
-        emailCacheService.saveEmail(request.user.internalId, EmailStatus(formData.value)).map {
+        emailCacheService.save(request.user.internalId, CachedData(formData.value, None)).map {
```

Comment:
It this situation class name "CachedData" seems to be more counter-intuitive. If it was something like EmailData or EmailDetails it would have more sense in the context. And that would apply also to Shailesh's comment

Reply:
yes, indeed. I will change this for sure in the final PR